client/client.go

```diff
@@ -0,0 +1,222 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"github.com/myteksi/hystrix-go/hystrix"
+	"http-echo/server"
+	"http-echo/server/config"
+	"io/ioutil"
+	"log"
+	"net/http"
+
+	//"io/ioutil"
+	//"net/http"
+	"sync"
+	"time"
+)
+
+var (
+	etcdClient       server.EtcdClient
+	serviceInstances ServiceInstances
+)
+
+func init() {
+	configPath := "./service.yml"
+	config := config.Loadconfig(configPath)
+	etcdClient = server.NewEtcdClient(config.Etcd.Endpoints, config.Etcd.Timeout)
+	serviceInstances = ServiceInstances{instances: map[string]ServiceInstance{}, endpoints: make([]string, 0), next: 0}
+	WatchServiceInstances(server.EtcServicePrefix, etcdClient, &serviceInstances)
+	serviceInstances.RemoveServiceInstanceBackground()
+}
+
+type ServiceInstance struct {
+	location         string
+	enable           bool
+	timeChangeStatus time.Time
+}
+
+type ServiceInstances struct {
+	instances map[string]ServiceInstance
+	mu        sync.Mutex
+	endpoints []string
```

Comment:
just wonder, why don't we define endpoints as map? then we can get rid of poor performance `removing item in array` logic?

Reply:
i just using list because i try using the round robin for loadbalance on pool of instances. My idea was using next property as index to get instance what will serve request. I will try to find another solution base on map to improve performance. Thank about your advice.